vowpalwabbit/global_data.h

```diff
@@ -382,8 +382,8 @@ struct vw
   vw_ostream trace_message;
 
   // Flag used when VW internally manages lifetime of options object.
-  bool should_delete_options = false;
-  VW::config::options_i* options;
+  bool should_release_options = false;

```

Comment:
This could probably be tracked at creation time of the unique_ptr with the deleter function?

Reply:
Not sure what you mean. This is in case we want to delete the options pointer outside of `all` object, so all should just release it.

Reply:
Added a comment at the spot where we'd need to do it.

Instead of using the `should_release_options` state, we can just make the deleter of the unique_ptr a noop in the cases where we would have called release. This also mitigates the case where an exception would have caused the object we're meant to release to be deleted.